drivers/input/input_pmw3610.c

```diff
@@ -202,6 +259,87 @@ static void pmw3610_motion_handler(const struct device *gpio_dev,
 	k_work_submit(&data->motion_work);
 }
 
+int pmw3610_set_resolution(const struct device *dev, uint16_t res_cpi)
+{
+	uint8_t val;
+	int ret;
+
+	if (!IN_RANGE(res_cpi, RES_MIN, RES_MAX)) {
+		LOG_ERR("res_cpi out of range: %d", res_cpi);
+		return -EINVAL;
+	}
+
+	ret = pmw3610_spi_clk_on(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = pmw3610_write_reg(dev, PWM3610_SPI_PAGE0, SPI_PAGE0_1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = pmw3610_read_reg(dev, PMW3610_RES_STEP, &val);
+	if (ret < 0) {
+		return ret;
+	}
+
+	val &= ~RES_STEP_RES_MASK;
+	val |= res_cpi / RES_STEP;
+
+	ret = pmw3610_write_reg(dev, PMW3610_RES_STEP, val);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = pmw3610_write_reg(dev, PWM3610_SPI_PAGE1, SPI_PAGE1_0);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = pmw3610_spi_clk_off(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+int pmw3610_force_awake(const struct device *dev, bool enable)
+{
+	uint8_t val;
+	int ret;
+
+	ret = pmw3610_read_reg(dev, PMW3610_PERFORMANCE, &val);
+	if (ret < 0) {
+		return ret;
+	}
+
+	val &= ~PERFORMANCE_FMODE_MASK;
+	if (enable) {
+		val |= PERFORMANCE_FMODE_FORCE_AWAKE;
+	} else {
+		val |= PERFORMANCE_FMODE_NORMAL;
+	}
+
+	ret = pmw3610_spi_clk_on(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = pmw3610_write_reg(dev, PMW3610_PERFORMANCE, val);
+	if (ret < 0) {
+		return ret;
```

Comment:
Maybe it would make sense to disable the `clk` if any of these fail?

Reply:
I don't think it does, if any of these fails the SPI bus is a write off, no sense trying to write more data on it.

Reply:
From the datasheet that I found there was not really a whole lot of information about the behaviour of the device. I assumed that maybe this would have some benefits in regards to power consumption. But yeah - fair enough :^)

Reply:
The datasheet is desperate, found so many gray areas and inconsistencies. Haven't tried to measure the effect of leaving this on but the whole description of what this bit does is so poor that I thought it was a leftover from some previous implementation of the device and and I originally tried to leave it off the driver entirely. Then I did some more tests specifically with The Force Awake[ns]™ mode and noticed that sometimes the device mode configuration was not applying correctly without that spi clock bit set, but it would by reading the register back or setting that bit on. So my impression is that this internally controls some logic for transferring over some configuration bit to the internal asic, or something like that. Anyway, this works for me, shrugs.